cmd/cluster-agent/subcommands/start/command.go

```diff
@@ -126,6 +126,10 @@ func Commands(globalParams *command.GlobalParams) []*cobra.Command {
 			// TODO: once the cluster-agent is represented as a component, and
 			// not a function (start), this will use `fxutil.Run` instead of
 			// `fxutil.OneShot`.
+
+			defaultforwarderParams := defaultforwarder.NewParamsWithResolvers()
+			defaultforwarderParams.DisableAPIKeyChecking()
```

Comment:
Can't we do something like:
`defaultforwarderParams := defaultforwarder.NewParamsWithResolvers().DisableAPIKeyChecking()`
by changing the return type of `DisableAPIKeyChecking()` ?

Reply:
I have updated the code to use the optional pattern.